official/recommendation/data_async_generation.py

```diff
@@ -455,17 +455,25 @@ def _parse_flagfile():
       sys.exit()
     time.sleep(1)
   tf.logging.info("flagfile found.")
-  # This overrides FLAGS with flags from flagfile.
-  flags.FLAGS([__file__, "--flagfile", flagfile])
+  # `flags` module opens `flagfile` with `open`, which does not work on
+  # google cloud storage etc.
+  with tf.gfile.Open(flagfile, "r") as f:
+    # This overrides FLAGS with flags from flagfile.
+    flags.FLAGS([__file__] + f.read().splitlines())
+  return flagfile
 
 
 def main(_):
   global _log_file
-  _parse_flagfile()
+  flagfile = _parse_flagfile()
 
   redirect_logs = flags.FLAGS.redirect_logs
   cache_paths = rconst.Paths(
       data_dir=flags.FLAGS.data_dir, cache_id=flags.FLAGS.cache_id)
+  # Rename flagfile so the next run at the same data_dir will still work.
```

Comment:
This seems very brittle. Currently multiple jobs can run from the same data dir, because everything temporary is in a unique cache directory. I think we want to preserve that property. (Apologies for not catching this in the review of the last PR)

Reply:
My thinking is that the cache id in ncf main would default to none. If none, generate on the fly as we do now. That way the user need not bother with it unless they need it to synchronize across workers.